Base.rte/Actors/Shared/Scripts/SelfHeal.lua

```diff
@@ -0,0 +1,115 @@
+				end
+				for _, arm in pairs(arms) do
+					if arm then
+						if self.healing.part.ID ~= arm.ID then
+							local offset = SceneMan:ShortestDistance(arm.JointPos, self.healing.wound.Pos, SceneMan.SceneWrapsX) + Vector(-math.max(self.healing.wound.Radius - 2, 1), 0):RadRotate(math.pi * 2 * timerRatio);
+							arm.HandPos = arm.JointPos + Vector(offset.X, offset.Y):SetMagnitude(math.min(offset.Magnitude, arm.MaxLength));
+						else
+							arm.HandPos = arm.JointPos + Vector(arm.MaxLength * (0.5 + math.sin(timerRatio * math.pi) * 0.5) * self.FlipFactor, 0):RadRotate(self.RotAngle);
+						end
+					end
+				end
+				self.healing.wound.Scale = math.min(self.healing.wound.Scale, self.healing.wound.Scale * (1 - timerRatio));
+			end
+		else	
+			self.healing = nil;
+		end
+		if self.EquippedItem or self.EquippedBGItem or controller:IsState(Controller.BODY_JUMP) or (self.FGArm or self.BGArm) == nil then
+			self.healing = nil;
+			self:RemoveNumberValue("SelfHeal");
+		else
+			for state = 1, 15 do
+				controller:SetState(state, false);
+			end
+			controller:SetState(Controller.BODY_CROUCH, true);
+		end
+	elseif self:NumberValueExists("SelfHeal") then
+		self.healing = {};
+		local priority = self:GetWoundCount(false, false, false) * self.DamageMultiplier;
+		self.healing.part = self;
+		for wound in self.Wounds do
+			self.healing.wound = wound;
+		end
+		for attachable in self.Attachables do
+			if attachable.DamageMultiplier > 0 then
+				local woundCount = attachable:GetWoundCount(false, false, false) * attachable.DamageMultiplier;
+				if woundCount > priority then
+					priority = woundCount;
+					self.healing.part = attachable;
+					for wound in attachable.Wounds do
+						self.healing.wound = wound;
+						break;
+					end
+				end
+				--Find damage-transferring sub-attachables as well
+				for attAttachable in attachable.Attachables do
+					if attAttachable.DamageMultiplier > 0 then
+						local woundCount = attAttachable:GetWoundCount(false, false, false) * attAttachable.DamageMultiplier;
+						if woundCount > priority then
+							priority = woundCount * attachable.DamageMultiplier;
+							self.healing.part = attAttachable;
+							for wound in attAttachable.Wounds do
+								self.healing.wound = wound;
+								break;
+							end
+						end
+					end
+				end
+			end
+		end
+		if self.healing.part and self.healing.wound then
+			self.healing.timer = Timer();
+			self.healing.delay = self.baseHealDelay * math.sqrt(self.healing.wound.Radius);
+			self:UnequipArms();
+		else
+			self.healing = nil;
+			controller:SetState(Controller.WEAPON_CHANGE_PREV, true);
+			self:RemoveNumberValue("SelfHeal");
+		end
+	end
+end
+
+function WhilePieMenuOpen(self, openedPieMenu)
+	openedPieMenu:GetFirstPieSliceByPresetName("Self Heal").Enabled = self.WoundCount > 0 and (self.FGArm or self.BGArm) ~= nil;
+end
+
+function SelfHeal(pieMenu, pieSlice, pieMenuOwner)
```

Comment:
This (and any other piemenu functions) needs to be reorganized cause I changed the arg order again lol
I believe it should become `pieMenuOwner, pieMenu, pieSlice`